### PR TITLE
#1084 fixing board page in firefox

### PR DESCRIPTION
**Pull request by krzyk**
#1084 
- fixed check in board.xsl for funds (the cause of the error)
- added tests to guard the fix

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1158](https://github.com/zerocracy/farm/issues/1158) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1158](https://github.com/zerocracy/farm/issues/1158) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1661404961); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/zerocracy/tk/TkBoardTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        catalog.add(project.pid(), "2017/02/AAAABBBBC/");
+        catalog.link(
+            project.pid(),
+            "github",
+            repo.coordinates().repo()
+        );
+        catalog.publish(project.pid(), true);
+        MatcherAssert.assertThat(
+            this.firefoxView(farm),
+            Matchers.containsString(
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/TkBoardTest.java]: @krzyk is it possible to use `XhtmlMatchers` here?

File: src/test/java/com/zerocracy/tk/TkBoardTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            new Ledger.Transaction(
+                cash,
+                "assets", "cash",
+                "income", "sponsor",
+                "There is some funding just arrived"
+            )
+        );
+        MatcherAssert.assertThat(
+            this.firefoxView(farm),
+            Matchers.containsString(
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/TkBoardTest.java]: @krzyk is it possible to use `XhtmlMatchers` here?

File: src/test/java/com/zerocracy/tk/TkBoardTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                "title=\"The project has no funds, you will work for free\""
+            )
+        );
+    }
+
+    private String firefoxView(final Farm farm)
+        throws IOException {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/TkBoardTest.java]: @krzyk let's move this to the previous line

File: src/test/java/com/zerocracy/tk/TkBoardTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                "assets", "cash",
+                "income", "sponsor",
+                "There is some funding just arrived"
+            )
+        );
+        MatcherAssert.assertThat(
+            this.firefoxView(farm),
+            Matchers.containsString(
+                String.format(
+                    "title=\"The project is funded\">%s</", cash.toString()
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/TkBoardTest.java]: @krzyk no need to use `toString()` with `%s` conversion

File: src/test/java/com/zerocracy/tk/TkBoardTest.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        MatcherAssert.assertThat(
+            this.firefoxView(farm),
+            XhtmlMatchers.hasXPaths(
+                new JoinedText(
+                    "/",
+                    "/xhtml:html/xhtml:body/xhtml:section/xhtml:article",
+                    "xhtml:table/xhtml:tbody/xhtml:tr[1]/xhtml:td[1]",
+                    "xhtml:span",
+                    // @checkstyle LineLength (1 line)
+                    "xhtml:span[@title = 'The project has no funds, you will work for free']"
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/TkBoardTest.java]: @krzyk can we use `//` selection to just look for the span and shorten the overall xpath? Selecting very specific elements from root seems fragile here.

File: src/test/java/com/zerocracy/tk/TkBoardTest.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            this.firefoxView(farm),
+            XhtmlMatchers.hasXPaths(
+                new JoinedText(
+                    "/",
+                    "/xhtml:html/xhtml:body/xhtml:section/xhtml:article",
+                    "xhtml:table/xhtml:tbody/xhtml:tr[1]/xhtml:td[1]",
+                    "xhtml:span",
+                    String.format(
+                        // @checkstyle LineLength (1 line)
+                        "xhtml:span[@title = 'The project is funded' and . = '%s']",
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/TkBoardTest.java]: @krzyk can we use `//` selection to just look for the span and shorten the overall xpath? Selecting very specific elements from root seems fragile here.

**Comment by krzyk** [comment]
@carlosmiranda I shortened the xpaths, have a look

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1158](https://github.com/zerocracy/farm/issues/1158) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by krzyk** [comment]
@g4s8 could you release the new version with this fix?

**Comment by g4s8** [comment]
@rultor release, tag=`0.22.7`

**Pull request merged**